clustering/singleton/extension/src/main/java/org/wildfly/extension/clustering/singleton/deployment/SingletonDeploymentProcessor.java

```diff
@@ -55,13 +56,14 @@ public void deploy(DeploymentPhaseContext context) throws DeploymentUnitProcessi
                 if (unit.putAttachment(Attachments.DEPLOYMENT_UNIT_PHASE_BUILDER, new SingletonDeploymentUnitPhaseBuilder(support, policy)) == null) {
                     SingletonLogger.ROOT_LOGGER.singletonDeploymentDetected(policy);
                     ServiceController<?> controller = context.getServiceRegistry().getRequiredService(unit.getServiceName());
-                    controller.addListener(new AbstractServiceListener<Object>() {
+                    controller.addListener(new LifecycleListener() {
```

Comment:
Because we're being extra careful about metaspace usage, please modify this such that SingletonDeploymentProcessor implements LifecycleListener, and registration becomes: controller.addListener(this);

Reply:
@pferraro sure